pennylane/beta/tapes/tape.py

```diff
@@ -233,6 +233,9 @@ def __init__(self, name=None):
         parameter indices (in the order they appear on the tape), and values are a
         dictionary containing the corresponding operation and operation parameter index."""
 
+        self._returns_state = False
```

Comment:
I wasn't able to see where this ever gets set to something else. It this an attribute that is set by client code (e.g., a QNode)?

Reply:
Nevermind, just spotted it :man_facepalming:

Reply:
I can't quite put my finger on it, but something is bothering me about this implementation :thinking:

Reply:
Note that it's updated to `self._state_measurement` in the new approach that doesn't require users to specify the wires - this objects is the measurement process itself that we later assign wires to.

However, the underlying implementation is still the same :thinking:, i.e., we set `self._state_measurement` when running `_process_queue()` and then later on access it in the `jacobian()` and `execute_device()` methods.

Reply:
Awesome, I've tried to minimize this logic in the tape (e.g. by pushing it out to the device).
One remaining thing is that the jacobian still checks if the return type is state - I'm not sure how we'd get around that.